pkg/registry/core/pod/storage/storage.go

```diff
@@ -241,6 +243,21 @@ func (r *BindingREST) setPodHostAndAnnotations(ctx context.Context, podUID types
 		for k, v := range annotations {
 			pod.Annotations[k] = v
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.PodTopology) {
+			// If any labels are present on the Binding, copy them across to the Pod.
+			if len(labels) > 0 {
+				if pod.Labels == nil {
+					pod.Labels = make(map[string]string)
+				}
+				for k, v := range labels {
+					if _, ok := pod.Labels[k]; ok {
+						// don't overwrite labels that are already present on the Pod
```

Comment:
Silent failure doesn't feel right?  How could this happen, and what does it mean if we hit it?  I agree we shouldn't overwrite existing labels in general, but how about WRT the known topology.k8s.io/* labels?  We own those

Reply:
Nothing stops a user setting these label keys on a Pod object explicitly themselves. It'd be weird, but I don't want to break some class of workload that happens to use/rely on/somehow work with an explicitly set label.